SharedSource/UnitTests/Reflection/TypeExtensionsTest.cs

```diff
@@ -0,0 +1,170 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using System.Reflection;
+using NUnit.Framework;
+using Remotion.Reflection;
+
+#nullable enable
+// ReSharper disable once CheckNamespace
+namespace Remotion.UnitTests.Reflection
+{
+  [TestFixture]
+  public class TypeExtensionsTest
+  {
+    [Test]
+    public void GetAssemblyQualifiedNameSafe_SystemType_AssemblyQualifiedName ()
+    {
+      var typeArgument = typeof (string);
+      const string expectedAssemblyQualifiedNamePrefix = "System.String,";
```

Comment:
Since we don't use the string more than once, it would be preferred to inline the value. That way, there's less indirection when trying to understand the test. This typically improves readability and makes copy&paste errors more obvious, e.g. when there's more than one assertion with a string, for instance when testing AQN and FullName in the same test.

I think I we should do this in all the tests here.

Reply:
I agree and have implemented your suggestion, but see the comment regarding `string.format` for my reasoning.